src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -0,0 +1,294 @@
+
+            // Again, for now only a block at a time
+            if (inputCount == 0)
+            {
+                return (Array.Empty<byte>());
+            }
+            else
+            {
+                char[] temp = new char[4];
+                Convert.ToBase64CharArray(inputBuffer, inputOffset, inputCount, temp, 0);
+                byte[] tempBytes = Encoding.ASCII.GetBytes(temp);
+                return (tempBytes);
+            }
+        }
+
+        // Must implement IDisposable, but in this case there's nothing to do.
+
+        public void Dispose()
+        {
+            Clear();
+        }
+
+        public void Clear()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing) { }
+
+        ~ToBase64Transform()
+        {
+            // A finalizer is not necessary here, however since we shipped a finalizer that called
+            // Dispose(false) in desktop v2.0, we need to keep it in case any existing code had subclassed
+            // this transform and expects to have a base class finalizer call its dispose method.
+            Dispose(false);
+        }
+    }
+
+    public class FromBase64Transform : ICryptoTransform
+    {
+        private byte[] _inputBuffer = new byte[4];
+        private int _inputIndex;
+        private FromBase64TransformMode _whitespaces;
+
+        public FromBase64Transform() : this(FromBase64TransformMode.IgnoreWhiteSpaces) { }
+        public FromBase64Transform(FromBase64TransformMode whitespaces)
+        {
+            _whitespaces = whitespaces;
+            _inputIndex = 0;
+        }
+
+        // Converting from Base64 generates 3 bytes output from each 4 bytes input block
+        public int InputBlockSize
+        {
+            get { return (1); }
+        }
+
+        public int OutputBlockSize
+        {
+            get { return (3); }
+        }
+
+        public bool CanTransformMultipleBlocks
+        {
+            get { return (false); }
+        }
+
+        public virtual bool CanReuseTransform
+        {
+            get { return (true); }
+        }
+
+        public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
+        {
+            // Do some validation
+            if (inputBuffer == null) throw new ArgumentNullException(nameof(inputBuffer));
+            if (inputOffset < 0) throw new ArgumentOutOfRangeException(nameof(inputOffset), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (inputCount < 0 || (inputCount > inputBuffer.Length)) throw new ArgumentException(SR.Argument_InvalidValue);
+            if ((inputBuffer.Length - inputCount) < inputOffset) throw new ArgumentException(SR.Argument_InvalidOffLen);
+
+            if (_inputBuffer == null)
+                throw new ObjectDisposedException(null, SR.ObjectDisposed_Generic);
+
+            byte[] temp = new byte[inputCount];
```

Comment:
We overwrite temp below in the if block.  The allocation should be moved to the else block.